executor/executor.go

```diff
@@ -100,57 +99,6 @@ func (executor *Executor) backoff(
 	)
 }
 
-// executePendingDecisions pulls decisions pending in execution status to execute again
-// decisions can stuck in pending status if the it crashes while there are few decisions queued for execution
```

Comment:
from what I understand from the docstring, this is needed to avoid having decisions stuck in pending status if the agent crashes before they're executed.

Reply:
in the new approach if the status of the job is inprogress we will keeping pulling it, this is an old approach